hi-ml-cpath/src/health_cpath/utils/report_utils.py

```diff
@@ -237,3 +258,21 @@ def collect_class_info(metrics_df: pd.DataFrame) -> Tuple[int, List[str]]:
         class_names = [name.lstrip() for name in class_names[1:-1].replace("'", "").split(',')]
     class_names = validate_class_names(class_names=class_names, n_classes=num_classes)
     return (num_classes, list(class_names))
+
+
+def collect_epoch_info(metrics_df: pd.DataFrame) -> Dict[int, int]:
+    """
+    Get the maximum number of epochs for each round from metrics dataframe
+    :param metrics_df: Metrics dataframe, as returned by :py:func:`collect_hyperdrive_metrics()` and
+        :py:func:`~health_azure.aggregate_hyperdrive_metrics()`.
+    :return: Number of classes and list of class names
+    """
+    max_epochs_dict = {}
+    for i in metrics_df.columns:
+        hyperparams = metrics_df[i][AMLMetricsJsonKey.HYPERPARAMS]
+        hyperparams_name = hyperparams[AMLMetricsJsonKey.NAME]
+        hyperparams_value = hyperparams[AMLMetricsJsonKey.VALUE]
```

Comment:
[Optional] One suggestion here is to convert `hyperparams` into a dictionary, like `dict(zip(hyperparams[NAME], hyperparams[VALUE]))`. This can then be made into a more generic and reusable function that returns all hyperparameters for each child run. What do you think?

Reply:
This is a good idea. I have now created a new function to get the hyperparameters, since these are required in `get_max_epochs` and `collect_class_info`